src/libstd/keyword_docs.rs

```diff
@@ -130,8 +134,8 @@ mod crate_keyword { }
 ///
 /// Enums in Rust are similar to those of other compiled languages like C, but have important
 /// differences that make them considerably more powerful. What Rust calls enums are more commonly
-/// known as Algebraic Data Types if you're coming from a functional programming background, but
-/// the important part is that data can go with the enum variants.
+/// known as Algebraic Data Types if you're coming from a functional programming background. The
```

Comment:
Continuing on... I would simply make "Algebraic Data Types" a link pointing to https://en.wikipedia.org/wiki/Algebraic_data_type. E.g. [Algebraic Data Types](https://en.wikipedia.org/wiki/Algebraic_data_type).

While most people can find that by using google, effective documentation should provide such material directly instead of having the user go through indirections. The wiki page also has useful links that can give users more in-depth explanations in perhaps more familiar syntax.